agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4440,32 +4440,41 @@ beerocks::message::sWifiChannel slave_thread::channel_selection_select_channel()
             continue;
         }
         for (uint8_t i = 0; i < beerocks::message::SUPPORTED_CHANNELS_LENGTH; i++) {
-            auto channel         = hostap_params.supported_channels[i];
-            auto operating_class = wireless_utils::get_operating_class_by_channel(
-                channel.channel, static_cast<beerocks::eWiFiBandwidth>(channel.channel_bandwidth));
+            auto channel        = hostap_params.supported_channels[i];
+            auto bw             = static_cast<beerocks::eWiFiBandwidth>(channel.channel_bandwidth);
+            auto center_channel = wireless_utils::get_5g_center_channel(channel.channel, bw, true);
+            auto operating_class =
+                wireless_utils::get_operating_class_by_channel(channel.channel, center_channel, bw);
 
             // Skip channels from other operating classes
             if (operating_class != preference.oper_class) {
                 continue;
             }
             // Skip DFS channels
-            if (channel.is_dfs_channel) {
+            if (hostap_params.supported_channels[i].is_dfs_channel) {
                 LOG(DEBUG) << "Skip DFS channel " << int(channel.channel);
                 continue;
             }
             auto is_restricted =
                 std::find_if(preference.channels.begin(), preference.channels.end(),
                              [&](const beerocks::message::sWifiChannel &wifi_channel) {
-                                 return (channel.channel == wifi_channel.channel);
+                                 // operating classes 128,129,130 use center channel **unlike the other classes** (See Table E-4 in 802.11 spec)
+                                 if (operating_class == 128 || operating_class == 129 ||
+                                     operating_class == 130) {
+                                     return center_channel == wifi_channel.channel;
+                                 } else {
+                                     return (channel.channel == wifi_channel.channel);
```

Comment:
a bit confusing , consider changing to something like wifi_chan_params.channel (nitpick)

Reply:
`channel.channel` is fine IMO and it was that before this PR, so NACK.